Gems/RecastNavigation/Code/CMakeLists.txt

```diff
@@ -56,6 +56,24 @@ set_property(TARGET DetourCrowd PROPERTY FOLDER ${RECAST_FETCHCONTENT_FOLDER})
 set_property(TARGET DetourTileCache PROPERTY FOLDER ${RECAST_FETCHCONTENT_FOLDER})
 set_property(TARGET Recast PROPERTY FOLDER ${RECAST_FETCHCONTENT_FOLDER})
 
+# Export external recast navigation targets for installers.
+set_property(DIRECTORY APPEND PROPERTY O3DE_SUBDIRECTORY_INSTALL_CODE [[
+    include(CMakePrintHelpers)
+
+    add_library(RecastNavigation::Detour STATIC IMPORTED)
+    set_target_properties(RecastNavigation::Detour PROPERTIES IMPORTED_LOCATION "${CMAKE_ARCHIVE_OUTPUT_DIRECTORY}/Detour.lib")
```

Comment:
This seems very windows specific, even though this works on other operating systems currently as-is.  
Maybe theres a more CMake-y way to do this that invovles asking the properties of the fetchcontent thing for the final locations?  Or at least using 
```
${CMAKE_ARCHIVE_OUTPUT_DIRECTORY}/${CMAKE_STATIC_LIBRARY_PREFIX}Detour${CMAKE_STATIC_LIBRARY_SUFFIX}"
```
instead of ".lib"?  Theres also the debug postfix (but I guess installers don't include debug so it doesnt' enter the picture)

On this mac it fetched and built like this:
(buildfolder)/lib/debug/libDetour-d.a
(buildfolder)/lib/profile/libDetour.a

Reply:
Thanks, fixed for profile installers. 

I will address debug libraries as I address install location of recast external targets.
https://github.com/o3de/o3de/issues/10077